karton/core/backend.py

```diff
@@ -291,6 +292,8 @@ def delete_task(self, task: Task) -> None:
         :param task: Task object
         """
         self.redis.delete(f"{KARTON_TASK_NAMESPACE}:{task.uid}")
+        for _, resource in task.iterate_resources():
```

Comment:
I'm getting a bit stressed here what happens if exception gets thrown somewhere in the middle.

Before we were "reasonably robust", i.e. crash at any point wouldn't leave the system in indeterminate state, because GC could clean it up afterwards - worst case a single task would be lost.

But now, if exception messes up reference counters, we may leave tasks hanging forever (or other cursed things).

I don't think there's a way to add a proper error handling here (if network goes down in the middle of reference decrementing, all bets are off). But at least I think we could try to do this whole operation (almost) atomically using redis pipelines: https://redis-py-cluster.readthedocs.io/en/stable/pipelines.html